fs/ignorefs/ignorefs.go

```diff
@@ -52,12 +56,66 @@ type ignoreDirectory struct {
 	fs.Directory
 }
 
+func correctCacheDirSignature(ctx context.Context, f fs.File) (bool, error) {
```

Comment:
Given the comment in `repo.OpenWithConfig`, it may make sense to move this function to `repo` and export it instead.

Reply:
```suggestion
func isCorrectCacheDirSignature(ctx context.Context, f fs.File) (bool, error) {
```